lightning/src/ln/channelmonitor.rs

```diff
@@ -1766,319 +1445,83 @@ impl ChannelMonitor {
-					let mut inputs_desc = Vec::new();
-					let mut inputs_info = Vec::new();
-
-					macro_rules! sign_input {
-						($sighash_parts: expr, $input: expr, $amount: expr, $preimage: expr) => {
-							{
-								let (sig, redeemscript, htlc_key) = match self.key_storage {
-									Storage::Local { ref htlc_base_key, .. } => {
-										let htlc = &per_commitment_option.unwrap()[$input.sequence as usize].0;
-										let redeemscript = chan_utils::get_htlc_redeemscript_with_explicit_keys(htlc, &a_htlc_key, &b_htlc_key, &revocation_pubkey);
-										let sighash = hash_to_message!(&$sighash_parts.sighash_all(&$input, &redeemscript, $amount)[..]);
-										let htlc_key = ignore_error!(chan_utils::derive_private_key(&self.secp_ctx, revocation_point, &htlc_base_key));
-										(self.secp_ctx.sign(&sighash, &htlc_key), redeemscript, htlc_key)
-									},
-									Storage::Watchtower { .. } => {
-										unimplemented!();
-									}
-								};
-								$input.witness.push(sig.serialize_der().to_vec());
-								$input.witness[0].push(SigHashType::All as u8);
-								$input.witness.push($preimage);
-								$input.witness.push(redeemscript.clone().into_bytes());
-								(redeemscript, htlc_key)
-							}
-						}
-					}
-
-					for (idx, &(ref htlc, _)) in per_commitment_data.iter().enumerate() {
+					// Then, try to find htlc outputs
+					for (_, &(ref htlc, _)) in per_commitment_data.iter().enumerate() {
 						if let Some(transaction_output_index) = htlc.transaction_output_index {
 							let expected_script = chan_utils::get_htlc_redeemscript_with_explicit_keys(&htlc, &a_htlc_key, &b_htlc_key, &revocation_pubkey);
 							if transaction_output_index as usize >= tx.output.len() ||
 									tx.output[transaction_output_index as usize].value != htlc.amount_msat / 1000 ||
 									tx.output[transaction_output_index as usize].script_pubkey != expected_script.to_v0_p2wsh() {
-								return (txn_to_broadcast, (commitment_txid, watch_outputs), spendable_outputs); // Corrupted per_commitment_data, fuck this user
-							}
-							if let Some(payment_preimage) = self.payment_preimages.get(&htlc.payment_hash) {
-								if htlc.offered {
-									let input = TxIn {
-										previous_output: BitcoinOutPoint {
-											txid: commitment_txid,
-											vout: transaction_output_index,
-										},
-										script_sig: Script::new(),
-										sequence: idx as u32, // reset to 0xfffffffd in sign_input
-										witness: Vec::new(),
-									};
-									if htlc.cltv_expiry > height + CLTV_SHARED_CLAIM_BUFFER {
-										inputs.push(input);
-										inputs_desc.push(if htlc.offered { InputDescriptors::OfferedHTLC } else { InputDescriptors::ReceivedHTLC });
-										inputs_info.push((payment_preimage, tx.output[transaction_output_index as usize].value, htlc.cltv_expiry));
-										total_value += tx.output[transaction_output_index as usize].value;
-									} else {
-										let mut single_htlc_tx = Transaction {
```

Comment:
I'm probably missing something, but I don't see where this code moved.

Reply:
It's L1321 as of 8603afb, aggregation now happens in OnchainTxHandler so there is no more a isolated vs multiple htlcs branch anymore.